Mathlib/Analysis/AbsoluteValue/Equivalence.lean

```diff
@@ -151,6 +151,128 @@ theorem exists_one_lt_lt_one_of_not_isEquiv {v w : AbsoluteValue R S} (hv : v.Is
+  refine ⟨c r, lt_of_lt_of_le (by linarith) (hr₁ r ?_), fun j hj ↦ ?_, by simpa [c, haw]⟩
+  · exact Finset.le_sup_dite_pos (p := fun j ↦ j = i) (f := fun _ _ ↦ r₁) (Finset.mem_univ _) rfl
+  · simpa using hrₙ j hj _ <| Finset.le_sup_dite_neg (fun j ↦ j = i) (Finset.mem_univ j) _
+
+/--
+- `v i, w`: absolute values on `R`.
+- `v i` is inequivalent to `v j` for all `j ≠ i` via the divergent point `a : R`.
+- `v i` is inequivalent to `w` via the divergent point `b : R`.
+- `1 < w a`.
+
+Then there is a common divergent point `k : R` causing both `v i` and `w` to be inequivalent to
+each `v j` for `j ≠ i`.
+-/
+private theorem exists_one_lt_lt_one_pi_of_one_lt (ha : 1 < v i a) (haj : ∀ j ≠ i, v j a < 1)
+    (haw : 1 < w a) (hb : 1 < v i b) (hbw : w b < 1) :
+    ∃ k : R, 1 < v i k ∧ (∀ j ≠ i, v j k < 1) ∧ w k < 1 := by
+  let c : ℕ → R := fun n ↦ 1 / (1 + a⁻¹ ^ n) * b
+  have hcᵢ : Tendsto (fun n ↦ v i (c n)) atTop (𝓝 (v i b)) := by
+    have : v i a⁻¹ < 1 := map_inv₀ (v i) a ▸ inv_lt_one_of_one_lt₀ ha
+    simpa [c] using (tendsto_div_one_add_pow_nhds_one this).mul_const (v i b)
+  have hcⱼ (j : ι) (hj : j ≠ i) : atTop.Tendsto (fun n ↦ v j (c n)) (𝓝 0) := by
+    have : 1 < v j a⁻¹ := map_inv₀ (v j) _ ▸
+      (one_lt_inv₀ <| (v j).pos fun h ↦ by linarith [map_zero (v _) ▸ h ▸ ha]).2 (haj j hj)
+    simpa [c] using (tendsto_div_one_add_pow_nhds_zero this).mul_const _
+  have hcₙ : atTop.Tendsto (fun n ↦ w (c n)) (𝓝 (w b)) := by
+    have : w a⁻¹ < 1 := map_inv₀ w _ ▸ inv_lt_one_of_one_lt₀ haw
+    simpa [c] using (tendsto_div_one_add_pow_nhds_one this).mul_const (w b)
+  simp_rw [OrderTopology.topology_eq_generate_intervals,
+    TopologicalSpace.tendsto_nhds_generateFrom_iff, mem_atTop_sets, Set.mem_preimage] at hcⱼ
+  choose r₁ hr₁ using Filter.eventually_atTop.1 <| Filter.Tendsto.eventually_const_lt hb hcᵢ
+  choose rₙ hrₙ using fun j hj ↦ hcⱼ j hj (.Iio 1) (by simpa using ⟨1, .inr rfl⟩) (by simp)
+  choose rN hrN using Filter.eventually_atTop.1 <| Filter.Tendsto.eventually_lt_const hbw hcₙ
+  let r := max (Finset.univ.sup fun j ↦ if h : j = i then r₁ else rₙ j h) rN
+  refine ⟨c r, hr₁ r ?_, fun j hj ↦ ?_, ?_⟩
+  · exact le_max_iff.2 <| .inl <|
+      Finset.le_sup_dite_pos (p := fun j ↦ j = i) (f := fun _ _ ↦ r₁) (Finset.mem_univ _) rfl
+  · exact hrₙ j hj _ <| le_max_iff.2 <| .inl <|
+      Finset.le_sup_dite_neg (fun j ↦ j = i) (Finset.mem_univ j) _
+  · exact hrN _ <| le_max_iff.2 (.inr le_rfl)
+
+open Fintype Subtype in
+/--
+If `v : ι → AbsoluteValue R S` is a finite collection of non-trivial and pairwise inequivalent
+absolute values, then for any `v i` there is some `a : R` such that `1 < v i a` while all other
+`v j a < 1`.
+-/
+theorem exists_one_lt_lt_one_pi_of_not_isEquiv (h : ∀ i, (v i).IsNontrivial)
+    (hv : Pairwise fun i j ↦ ¬(v i).IsEquiv (v j)) :
+    ∀ i, ∃ (a : R), 1 < v i a ∧ ∀ j ≠ i, v j a < 1 := by
+  let P (ι : Type u_3) [Fintype ι] : Prop := [DecidableEq ι] →
```

Comment:
```suggestion
  let P (ι : Type _) [Fintype ι] : Prop := [DecidableEq ι] →
```
Please don't use autogenerated names, `Type _` is there for this reason (it is sometimes slow, in that case just give a name to the universe of `ι`).